Enable optional float confs and update docs mentioning them

razajafri opened the pull request:
We are enabling the float confs and their respective docs that were turned off by default before 22.06

These are the confs that were set to true in this PR 

`spark.rapids.sql.castDecimalToFloat.enabled`
`spark.rapids.sql.castFloatToDecimal.enabled`
`spark.rapids.sql.castFloatToIntegralTypes.enabled`
`spark.rapids.sql.castFloatToString.enabled`
`spark.rapids.sql.castStringToFloat.enabled`

`spark.rapids.sql.csv.read.float.enabled`
`spark.rapids.sql.json.read.float.enabled`

`spark.rapids.sql.improvedFloatOps.enabled`

`spark.rapids.sql.variableFloatAgg.enabled`



fixes #5323 

Signed-off-by: Raza Jafri <rjafri@nvidia.com>

<!--

Thank you for contributing to RAPIDS Accelerator for Apache Spark!

Here are some guidelines to help the review process go smoothly.

1. Please write a description in this text box of the changes that are being
   made.

2. Please ensure that you have written units tests for the changes made/features
   added.

3. If you are closing an issue please use one of the automatic closing words as
   noted here: https://help.github.com/articles/closing-issues-using-keywords/

4. If your pull request is not ready for review but you want to make use of the
   continuous integration testing facilities please label it with `[WIP]`.

5. If your pull request is ready to be reviewed without requiring additional
   work on top of it, then remove the `[WIP]` label (if present).

6. Once all work has been done and review has taken place please do not add
   features or make changes out of the scope of those requested by the reviewer
   (doing this just add delays as already reviewed code ends up having to be
   re-reviewed/it is hard to tell what is new etc!). Further, please avoid
   rebasing your branch during the review process, as this causes the context
   of any comments made by reviewers to be lost. If conflicts occur during
   review then they should be resolved by merging into the branch used for
   making the pull request.

Many thanks in advance for your cooperation!

-->

razajafri commented:
@sameerz PTAL

sameerz commented on docs/get-started/getting-started-workload-qualification.md:
````diff
@@ -101,15 +101,11 @@ the driver logs with `spark.rapids.sql.explain=all`.
   ```scala
-   spark.conf.set("spark.rapids.sql.incompatibleOps.enabled", true)
-   spark.conf.set("spark.rapids.sql.variableFloatAgg.enabled", true)
    spark.conf.set("spark.rapids.sql.decimalType.enabled", true)
````
I believe removed spark.rapids.sql.decimalType.enabled in https://github.com/NVIDIA/spark-rapids/commit/de6d45492082ab107829cb810bb2312db7744c40#diff-12777fe3bbb26d1a892f120acf176b07af222570dafd19f039adc3f84d4c58e2L547 , and should remove it from this doc. 
```suggestion
```

sameerz commented on docs/get-started/getting-started-workload-qualification.md:
````diff
@@ -220,15 +216,11 @@ Throws:
    ```scala
-   spark.conf.set("spark.rapids.sql.incompatibleOps.enabled", true)
-   spark.conf.set("spark.rapids.sql.variableFloatAgg.enabled", true)
    spark.conf.set("spark.rapids.sql.decimalType.enabled", true)
````
Same comment as above about decimalType.enabled not being a valid option. 
```suggestion
```

sameerz reviewed (changes requested):
The following docs also refer to the changed configs and need to be updated:
### tuning-guide.md
`castStringToFloat.enabled` 
`castFloatToString.enabled`
`incompatibleOps.enabled`
`variableFloatAgg.enabled`

### compatibility.md
`castStringToFloat.enabled`
`castFloatToIntegralTypes.enabled`
`castFloatToString.enabled`
`incompatibleOps.enabled`
`variableFloatAgg.enabled`

### supported_ops.md
`castFloatToString.enabled`
`incompatibleOps.enabled`

razajafri commented:
build

razajafri commented:
> The following docs also refer to the changed configs and need to be updated:
> ### tuning-guide.md
> 
> `castStringToFloat.enabled` `castFloatToString.enabled` `incompatibleOps.enabled` `variableFloatAgg.enabled`

I have removed this entire section

> ### compatibility.md
> 
> `castStringToFloat.enabled` `castFloatToIntegralTypes.enabled` `castFloatToString.enabled` `incompatibleOps.enabled` `variableFloatAgg.enabled`

I had already updated the wordings of this doc. I don't think the variables should be removed from this doc as they still affect the compatibility

> ### supported_ops.md
> 
> `castFloatToString.enabled` `incompatibleOps.enabled`

I don't think this should be changed either as this is still important for the user to read

revans2 commented:
Can we rename this PR. "optional confs and updated docs" for what?  What is being enabled. All optional confs? Just floating point?

razajafri commented:
@revans2 Is this better now?

razajafri commented:
build

razajafri commented:
@sameerz I think I have addressed your concerns. PTAL

sameerz commented on docs/tuning-guide.md:
```diff
@@ -278,30 +278,6 @@ from the main [columnar batch size](#columnar-batch-size) setting.  Some transco
-The following configs all enable different types of incompatible operations that can improve
-performance.
-- [`spark.rapids.sql.variableFloatAgg.enabled`](configs.md#sql.variableFloatAgg.enabled) 
-- [`spark.rapids.sql.hasNans`](configs.md#sql.hasNans)
```
We have not removed `hasNans` yet, I think we should continue to mention it's purpose and how to remove, and that it can improve performance.  

Depends on https://github.com/NVIDIA/spark-rapids/issues/5320

razajafri replied:
Should we just leave this unchanged? These configs are on by default but doesn't change anything else

razajafri commented:
@sameerz I have added the section discussing the incompatible operations, the only change should be the default value. 

PTAL

razajafri commented:
build

razajafri commented:
It's complaining about 
```
ERROR: 1 dead links found!
[✖] https://cloud.google.com/sdk/install → Status: 503
```
but it's not dead

razajafri commented:
@sameerz do you have any more concerns on this PR?

razajafri commented:
Build
